src/thread/thread_manager.cpp

```diff
@@ -140,7 +140,7 @@ namespace cubthread
 
   entry_workpool *
   manager::create_worker_pool (size_t pool_size, size_t task_max_count, entry_manager *context_manager,
-			       std::size_t core_count, bool debug_logging)
+			       std::size_t core_count, bool debug_logging, bool pool_threads, cubperf::duration transition_period)
```

Comment:
Rename as wait_for_task_time or fade_out_time or something more descriptive than "transition".

cubperf::duration should be reserved to performance statistics. I think std::chrono::seconds is more appropriate.